### ISSUE TITLE
Nchan dynamic module compile error with gcc 10.x and clang 11.x

### ISSUE THREAD

**Issue by chrislea** [slact/nchan#603]
Hi. As far as I can tell, nchan 1.2.7 won't compile as a dynamic module on Ubuntu 20.10 Groovy Gorilla. Debugging this is I think past me, but the linker seems to be upset about something.

I am attaching a zip that contains two files. The `nchan_build_fail.sh` is a bash script that when run on a vanilla amd64 20.10 ec2 instance will install everything needed to build Nginx with Nchan, try to compile it, and subsequently fail. The `build_fail.txt` file is simply the end-to-end output of the bash script.

I have not seen the same problem with earlier versions of Ubuntu using the same build sources. The build fails using both `gcc` and `clang` in the same way.

Please let me know if I can provide any additional information.
[nchan_build_files.zip](https://github.com/slact/nchan/files/5755822/nchan_build_files.zip)

**Comment by chrislea** [comment]
After poking around a bit more, the link error occurs only with `gcc 10.x` or `clang 11.x`. These are the default compilers installed on Ubuntu 20.10, but the same errors occur on my Arch laptop using these same new compilers.

Ubuntu 20.10 has `gcc 9.x` and `clang 10.x` available in the standard repositories. I just tried building using those and things worked correctly. So something has changed with these newest compilers that makes the build fail.

I've updated the issue title to be more explicit.
